api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -165,6 +165,12 @@ private void updateDataAccessLevel(DbUser dbUser) {
     if (shouldUserBeRegistered(dbUser)) {
       addToRegisteredTierGroupIdempotent(dbUser);
       newDataAccessLevel = DataAccessLevel.REGISTERED;
+
+      // if this is the first time the user has completed registration, record it
+      // this starts the Free Tier Credits countdown clock
+      if (dbUser.getFirstRegistrationCompletionTime() == null) {
+        dbUser.setFirstRegistrationCompletionTime();
```

Comment:
I believe this will effectively autobackfill the timestamp for all registered users after the next 2FA sync cronjob runs following this code's release. Does that match your expectations?

Reply:
This is not something I had considered.

We should add this as a note to the audit/backfill ticket.  Thanks.